### PR TITLE
4cam functionally fixed

### PR DESCRIPTION
**Pull request by mahi97**
This closes #23

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ addons:
 
 before_install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update           ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install qt4      ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then sh mac-build.sh       ; fi
```
Comment (mani-monaj) [inline .travis.yml]: What is the problem with installing `qt4` using *brew*? I personally prefer the current method.

File: include/sslworld.h [hunk 2]
```diff
@@ -60,20 +60,25 @@ class SSLWorld : public QObject
     QList<SendingPacket*> sendQueue;
     char packet[200];
     char *in_buffer;    
+	void adjustOverlapPos(dReal &p, bool positive);
```
Comment (mani-monaj) [inline include/sslworld.h]: All the indentations are inconsistent, please fix.

File: src/sslworld.cpp [hunk 3]
```diff
@@ -1019,3 +989,130 @@ dReal rand0_1()
 {
     return (dReal) (rand()) / (dReal) (RAND_MAX);
 }
+
+void SSLWorld::adjustOverlapPos(dReal &p, bool positive) {
+	if(fabs(p) <= 0.30) {
```
Comment (mani-monaj) [inline src/sslworld.cpp]: `0.3` should be a parameter, preferably runtime configurable.

File: src/sslworld.cpp [hunk 4]
```diff
@@ -1019,3 +989,130 @@ dReal rand0_1()
...
+		double factor = 1-(double)fabs(p)/0.30;
+		p += factor*(positive? 0.02 : -0.02);
```
Comment (mani-monaj) [inline src/sslworld.cpp]: Same for `0.02`

File: .travis.yml (hunk 1 above)
Comment (mani-monaj) [inline .travis.yml]: As you can see it also broke the build on Travis.

Comment (mahi97) [inline .travis.yml]: the main problem is the brew removed qt4 package.
So i'm trying to install Qt4 by another way ...
actually i thinks migrating to Qt5 is a better way but it's need lots of work i guess

File: include/sslworld.h (hunk 2 above)
Comment (mahi97) [inline include/sslworld.h]: Okey i will fix them 👍

File: .travis.yml (hunk 1 above)
Comment (mahi97) [inline .travis.yml]: I tried to tap another repository to brew and install qt4, but i can't found any stable source.
i also tried this one  `brew install cartr/qt4/qt` and that wasn't successful too :(

**Comment by mahi97** [comment]
@mani-monaj 
I tried to download and compile source of qt4 but it's not worked.
any suggestions ?

**Comment by mahi97** [comment]
I close this PR and after i fix this issue i will send another one.

**Pull request closed without merging**